app/controllers/api/base_controller.rb

```diff
@@ -37,6 +37,16 @@ def verify_mfa_requirement
     render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
   end
 
+  def should_setup_mfa?
+    @api_key.user&.recommend_mfa_setup?
+  end
+
+  def response_with_mfa_warning(response)
+    return response unless should_setup_mfa? 
+    message = "\n\n[WARNING] For protection of your account and your gems, you are encouraged to set up multi-factor authentication at https://rubygems.org/multifactor_auth/new. Your account will be required have MFA enabled in the future."
```

Comment:
`Your account will be required have MFA enabled in the future`

We can specify the date with `require_mfa_at` https://github.com/Shopify/rubygems.org/blob/730a9ac1f1a51b9b0746ccc0ef480024453162e2/db/migrate/20220111011544_add_require_mfa_at_to_users.rb#L3